src/Home.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react'
+import {Link} from 'react-router-dom'
+import * as BooksAPI from "./BooksAPI"
+
+import Book from "./Book"
+
+class HomePage extends Component {
+  state = {
+    books: []
+  }
+  componentDidMount() {
+    BooksAPI.getAll().then(books => {
+      this.setState({ books })
+    })
+  }
+  changeShelf = (e, book) => {
+    const books = this.state.books
+    const shelf = e.target.value
+    book.shelf = shelf
+    this.setState({
+      books
```

Comment:
`books` isn't changed in the above code so it seems this isn't doing anything. You're basically getting it from the state on line 17, then setting it to itself here.

Reply:
removed)